src/hardware/openbench-logic-sniffer/protocol.c

```diff
@@ -509,25 +509,19 @@ SR_PRIV int ols_receive_data(int fd, int revents, void *cb_data)
 
 			/* Send the trigger. */
 			std_session_send_df_trigger(sdi);
-
-			/* Send post-trigger samples. */
-			packet.type = SR_DF_LOGIC;
-			packet.payload = &logic;
-			logic.length = (devc->num_samples * 4) - (devc->trigger_at_smpl * 4);
-			logic.unitsize = 4;
-			logic.data = devc->raw_sample_buf + devc->trigger_at_smpl * 4 +
-				(devc->limit_samples - devc->num_samples) * 4;
-			sr_session_send(sdi, &packet);
-		} else {
-			/* no trigger was used */
-			packet.type = SR_DF_LOGIC;
-			packet.payload = &logic;
-			logic.length = devc->num_samples * 4;
-			logic.unitsize = 4;
-			logic.data = devc->raw_sample_buf +
-				(devc->limit_samples - devc->num_samples) * 4;
-			sr_session_send(sdi, &packet);
 		}
+
+		/* Send post-trigger / all captured samples. */
+		int num_pre_trigger_samples = devc->trigger_at_smpl == CTX_NO_TRIGGER
+			? 0 : devc->trigger_at_smpl;
```

Comment:
All this code looks correct. I got one question, though: Would it be possible to define CTX_NO_TRIGGER to 0 instead of -1? Basically meaning that CTX_NO_TRIGGER could go then. I think it is quite readable: 'trigger_at_sample = 0'  means no trigger. Then you could drop 'num_pre_trigger_samples' here completely and other calculations later will also become simpler. On top, the variables could be changed to unsigned int. I think it will also work for the later introduced 'trigger_rle_at_smpl_from_end' but I may be missing something. Do you think it is possible and benefitial?

Reply:
I think it's possible for the general case, but e.g. in the case of erroneous input, it leads
to confusion: If the first full sample received has the RLE mark, `trigger_rle_at_smpl_from_end` will be set to `cnt_samples` := 0.
Thus, I'd prefer to keep the out-of-band signal `OLS_NO_TRIGGER` at -1, which is outside of the valid value range.

Reply:
Okay, makes sense.